subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPluginExtension.java

```diff
@@ -45,7 +45,7 @@
     private final JacocoAgentJar agent;
 
     private String toolVersion;
-    private File reportsDir;
+    private Provider<File> reportsDir;
```

Comment:
It would be nice to add a type to represent a configurable file or directory that is-a or has-a `Provider<File>`, like `FileCollection` is-a or has-a `Provider<Set<File>>`.

This type can then have methods so you can do things like `def report = reportsDir.file("my-report.html")`